js_modules/dagster-ui/packages/ui-core/src/assets/catalog/useAssetCatalogGroupAndSortBy.tsx

```diff
@@ -0,0 +1,421 @@
+    decode: useCallback((qs: ParsedQs) => {
+      if (qs.groupBy && GROUP_BY.includes(qs.groupBy as AssetHealthGroupBy)) {
+        return qs.groupBy as AssetHealthGroupBy;
+      }
+      return AssetHealthGroupBy.health_status;
+    }, []),
+    encode: useCallback((b: AssetHealthGroupBy) => ({groupBy: b}), []),
+  });
+
+  const grouped: Record<string, Grouped<any, any>> = useMemo(() => {
+    switch (groupBy) {
+      case AssetHealthGroupBy.code_location:
+        return groupByAttribute({
+          liveDataByNode: assetsByAssetKey,
+          getAttributes: ({key}) => {
+            const asset = assetsByAssetKey.get(tokenForAssetKey(key));
+            const repo = asset?.definition?.repository;
+            return [repo ? buildRepoPathForHuman(repo.name, repo.location.name) : NONE_KEY];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <AttributeStatusHeaderRow
+                {...props}
+                text={props.group}
+                groupBy={AssetHealthGroupBy.code_location}
+                key={props.group}
+                noneGroup="No code location"
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.group:
+        return groupByAttribute({
+          liveDataByNode: assetsByAssetKey,
+          getAttributes: ({key}) => {
+            const asset = assetsByAssetKey.get(tokenForAssetKey(key));
+            return [asset?.definition?.groupName ?? 'default'];
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <AttributeStatusHeaderRow
+                {...props}
+                text={props.group}
+                groupBy={AssetHealthGroupBy.group}
+                key={props.group}
+                noneGroup="No group"
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.owner:
+        return groupByAttribute({
+          liveDataByNode: assetsByAssetKey,
+          getAttributes: ({key}) => {
+            const asset = assetsByAssetKey.get(tokenForAssetKey(key));
+            return (
+              asset?.definition?.owners.map((owner) =>
+                'email' in owner ? owner.email : owner.team,
+              ) ?? [NONE_KEY]
+            );
+          },
+          renderGroupHeader: (props) => {
+            return (
+              <AttributeStatusHeaderRow
+                {...props}
+                text={props.group}
+                groupBy={AssetHealthGroupBy.owner}
+                key={props.group}
+                noneGroup="No owner"
+              />
+            );
+          },
+        });
+      case AssetHealthGroupBy.kind:
+        return groupByAttribute({
+          liveDataByNode: assetsByAssetKey,
+          getAttributes: ({key}) => {
+            const asset = assetsByAssetKey.get(tokenForAssetKey(key));
+            if (asset?.definition?.computeKind || asset?.definition?.kinds?.length) {
```

Comment:
Hmm, we show Compute Kind and Kind as two separate fields on the asset overview definition page, and VirtualizedAssetRow only shows the definition.kind tags.  Maybe for consistency we don't include computeKind here, or we break out another GroupBy for compute kind?